src/librustc/middle/resolve_lifetime.rs

```diff
@@ -2421,7 +2421,8 @@ impl<'a, 'tcx> LifetimeContext<'a, 'tcx> {
 
             let help_name = if let Some(body) = parent {
                 let arg = &self.tcx.hir().body(body).arguments[index];
-                format!("`{}`", self.tcx.hir().hir_to_pretty_string(arg.original_pat().hir_id))
+                let original_pat = self.tcx.hir().original_pat_of_argument(arg);
+                format!("`{}`", self.tcx.hir().hir_to_pretty_string(original_pat.hir_id))
```

Comment:
Shouldn't this be using `simple_ident()` to avoid printing a potentially confusing/ambiguous pattern?

Reply:
Fixed.